include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -476,6 +477,27 @@ class alignment_file_input_format_REMOVEME<format_sam>
         }
     }
 
+
+    /*!\brief Transfer soft clipping information from the \p cigar_vector to \p sc_begin and \p sc_end.
+     * \param[in] cigar_vector The cigar information to parse for soft-clipping.
+     * \param[in,out] sc_begin The soft clipping at the beginning of the alignment to set.
+     * \param[in,out] sc_end   The soft clipping at the end of the alignment to set.
+     */
+    void transfer_soft_clipping_to(std::vector<cigar> const & cigar_vector, int32_t & sc_begin, int32_t & sc_end)
+    {
+        // check for soft clipping at the first two positions
```

Comment:
Because soft and hard clipping can only occur at the beginning or end of an aligned read.
It can be only hard or soft clipping (first position to check) but it can also occur both (first position hard clipping, second soft clipping).

Reply:
So there can be hard clipping followed by soft clipping. Is that it?

Reply:
yes